src/plinder/core/loader/loader.py

```diff
@@ -162,3 +162,59 @@ def get_model_input_files(
         if count >= max_num_sample:
             break
     return model_inputs
+
+
+# Example sampler function, this is for demonstration purposes,
+# users are advised to write a sampler that suit their need
+def get_diversity_samples(
```

Comment:
When this function is used rather for demonstration rather than for production, I think this should go directly into an example notebook.

Reply:
We wrote this quickly, so, I am not sure if it will be like this in production, but it does not look too bad, and I would like to avoid exposing too complicated functions to the end user in the notebooks.

There is a possibility that we or others would build the dataloader with more functionality in the future, but it is not a top priority in development for now.